components/org.wso2.carbon.identity.conditional.auth.functions.common/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/common/utils/CommonUtils.java

```diff
@@ -44,4 +50,53 @@ public static String getConnectorConfig(String key, String tenantDomain) throws
             throw new IdentityEventException("Error while getting connector configurations for property :" + key, e);
         }
     }
+
+    /**
+     * Check whether the given value is a secret alias.
+     *
+     * @param value Value to check
+     * @return True if the value is a secret alias, false otherwise
+     */
+    public static boolean isSecretAlias(String value) {
+
+        return value.startsWith(SECRET_PREFIX);
+    }
+
+    /**
+     * Resolve the secret alias to its actual value.
+     *
+     * @param alias Secret alias
+     * @return Actual secret value
+     */
+    public static String resolveSecretFromAlias(String alias) {
+
+        return alias.substring(SECRET_PREFIX.length());
+    }
+
+    /**
+     * Resolves the secret value, whether the input is a direct secret name or an alias.
+     * If the input is an alias, it first resolves the alias to the actual secret name and then fetches the secret value.
+     * If the input is not an alias, it returns the input directly, assuming it's already the resolved secret.
+     *
+     * @param secretOrAlias The secret name or alias.
+     * @return The input itself if it's not an alias, or the resolved secret value if it is an alias.
+     * @throws SecretManagementException If there is an error in resolving the secret from an alias.
+     */
+    public static String getResolvedSecret(String secretOrAlias) throws SecretManagementException {
+        if (StringUtils.isNotEmpty(secretOrAlias)) {
+            if (!isSecretAlias(secretOrAlias)) {
+                // If it's not an alias, return the input directly
+                return secretOrAlias;
+            } else {
+                // If it's an alias, strip the prefix to get the actual secret name and resolve it
+                String secretName = secretOrAlias.substring(SECRET_PREFIX.length());
+                ResolvedSecret responseDTO = FunctionsDataHolder.getInstance().getSecretConfigManager()
+                        .getResolvedSecret(SECRET_TYPE_CALL_CHOREO, secretName);
+                if (responseDTO != null) {
+                    return responseDTO.getResolvedSecretValue();
+                }
+            }
+        }
+        return null;
+    }
```

Comment:
Let's not have this segment merged as we are moving to a better secret resolving strategy

Reply:
Fixed with https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/157/commits/05c4f77e8d18ba911c3b98bc7e18ec3fb73dcd33